mealie/services/scraper/recipe_scraper.py

```diff
@@ -41,9 +44,23 @@ async def scrape(self, url: str, html: str | None = None) -> tuple[Recipe, Scrap
         raw_html = html or await safe_scrape_html(url)
         for scraper_type in self.scrapers:
             scraper = scraper_type(url, self.translator, raw_html=raw_html)
-            result = await scraper.parse()
 
-            if result is not None:
-                return result
+            try:
+                result = await scraper.parse()
+            except Exception:
+                self.logger.exception(f"Failed to scrape HTML with {scraper.__class__.__name__}")
+                result = None
+
+            if result is None or result[0] is None:
+                continue
+
+            recipe_result, extras = result
+            try:
+                recipe = cleaner.clean(recipe_result, self.translator)
+            except Exception:
```

Comment:
Is it worth catching the specific exception? If not maybe throw that exception in the log so we know what the Exception is.

Reply:
`logger.exception` logs the exception in addition to the log message, so the logs should show what it is

Reply:
You don't have to pass it in or anything, it just magically knows???

Reply:
Python is magic

```python
from logging import getLogger

logger = getLogger()

try:
    assert False
except Exception:
    logger.exception("bad exception")
```

```
bad exception
Traceback (most recent call last):
  File "...", line 6, in <module>
    assert False
AssertionError
```